### PR TITLE
Fix BOXAPIMultipartToJSONOperation not closing its output stream on t…

### PR DESCRIPTION
**Pull request by jpang1**
…he same runloop when operation is canceled.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jpang1 has signed the CLA. Thanks for the pull request!

**Comment by RicoYao** [comment]
lgtm. this makes much more sense than the old way.

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m [hunk 1]
```diff
@@ -590,6 +587,11 @@ - (void)writeDataToOutputStream
 {
     while ([self.outputStream hasSpaceAvailable])
     {
+        if (self.isCancelled) {
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m:590]: Anywhere else to handle this or are these 2 the only stream methods?

How do we make sure this is handled correctly if/when another method is added?

**Comment by boxtcarpel** [comment]
+1

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m (hunk 1 above)
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m:590]: These changes are both in one method. I believe there's only one stream delegate method for handling events in the runloop (handleEvent) so I don't think we need to worry about some other method.

**Pull request merged**